2017-10-28  Sidharth Singh Bambah  <bambah@lnxsrv09.seas.ucla.edu>

	* Patched the timespec.h file with if statements detecting if the return
	value should be positive or negative.

--- /u/ee/ugrad/bambah/Desktop/coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ /u/ee/ugrad/bambah/Desktop/coreutils-with-bug/lib/timespec.h	2017-10-28 21:01:43.436066000 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+	if(a.tv_sec > b.tv_sec)
+		return 1;
+	else if(a.tv_sec > b.tv_sec)
+		return 0;
+	else
+		return -1;	
 }
 
 # if ! HAVE_DECL_NANOSLEEP
